addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -132,6 +132,11 @@ def _cancel_invoice_edi_eta(self, invoice):
         access_data = self._l10n_eg_eta_get_access_token(invoice)
         if access_data.get('error'):
             return access_data
+        # Check current status. It may already be cancelled or rejected.
+        if invoice.l10n_eg_submission_number:
+            document_summary = self._l10n_eg_get_einvoice_document_summary(invoice)
+            if document_summary and document_summary[0].get('status') in ('Cancelled', 'Rejected'):
+                return {'success': True}
```

Comment:
While testing with the customer, we found that the ETA will delay cancelation from their end on a schedule.
Therefor, if the client tries to cancel in odoo during the pre-cancelation period, the invoice isn't in a cancelable state (and will be stuck in ETA 'error')

<img width="577" alt="image" src="https://github.com/user-attachments/assets/ed54888b-18c0-497b-a6ed-556e2889677d">

an example of the requests:
```py
[
    {
        "publicUrl": "REDACTED",
        "uuid": "REDACTED",
        "longId": "REDACTED",
        "internalId": "REDACTED",
        "typeName": "i",
        "typeVersionName": "1.0",
        "issuerId": "REDACTED",
        "issuerName": "REDACTED",
        "receiverId": "445584343",
        "receiverName": "REDACTED",
        "receiverType": 0,
        "issuerType": 0,
        "dateTimeIssued": "2024-07-22T00:00:00Z",
        "total": 25.65,
        "status": "Valid",
        "submissionUUID": "REDACTED",
        "dateTimeReceived": "2024-07-22T10:52:25.3054582Z",
        "totalSales": 22.5,
        "totalDiscount": 0.0,
        "netAmount": 22.5,
        "documentStatusReason": "بيانات المشتري خاطئة",
        "cancelRequestDate": "2024-07-22T11:00:10.0436131Z",
        "rejectRequestDate": None,
        "cancelRequestDelayedDate": "2024-07-23T11:00:10.0436131Z",
        "rejectRequestDelayedDate": None,
        "declineCancelRequestDate": None,
        "declineRejectRequestDate": None,
        "canbeCancelledUntil": "2024-07-29T10:52:25.3054582Z",
        "canbeRejectedUntil": "2024-07-29T10:52:25.3054582Z",
        "freezeStatus": {
            "frozen": False,
            "type": None,
            "scope": None,
            "actionDate": None,
            "auCode": None,
            "auName": None,
        },
        "lateSubmissionRequestNumber": None,
    }
]
```

Reply:
Is it the response to a `cancellation` request (https://sdk.invoicing.eta.gov.eg/einvoicingapi/02-cancel-document/) or a response to `get submission` request (https://sdk.invoicing.eta.gov.eg/einvoicingapi/09-get-submission/)?